src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1679,7 +1679,9 @@ protected function prepareRequestUri()
      */
     protected function prepareBaseUrl()
     {
-        $filename = basename($this->server->get('SCRIPT_FILENAME'));
+        $documentRoot = $this->server->get('DOCUMENT_ROOT');
```

Comment:
the document root should be retrieved only at the end, just before needing it, to avoid getting it when not needed